### PR TITLE
Add search order to assign to a warehouse

### PR DESCRIPTION
**Pull request by BB2A-Anthony**
# New Add search order to assign to a warehouse

Addition of a search filter for orders assigned to a warehouse.

### CODE REVIEW

File: htdocs/commande/list.php [hunk 1]
```diff
@@ -573,6 +576,14 @@
 		$moreforfilter .= $formother->select_categories('customer', $search_categ_cus, 'search_categ_cus', 1);
 	 	$moreforfilter .= '</div>';
 	}
+	if (!empty($conf->expedition->enabled)){
+		require_once DOL_DOCUMENT_ROOT.'/product/class/html.formproduct.class.php';
+		$formproduct = new FormProduct($db);
+		$moreforfilter .= '<div class="divsearchfield">';
+	 	$moreforfilter .= $langs->trans('Warehouse').': ';
+		$moreforfilter .= $formproduct->selectWarehouses($search_warehouse, 'search_warehouse', '', 1);
```
Comment (eldy) [inline htdocs/commande/list.php]: When did this variable $search_warehouse is used ?

**Review by BB2A-Anthony — COMMENTED** [review]
is not good?

File: htdocs/commande/list.php [hunk 2]
```diff
@@ -573,6 +576,14 @@
 		$moreforfilter .= $formother->select_categories('customer', $search_categ_cus, 'search_categ_cus', 1);
 	 	$moreforfilter .= '</div>';
 	}
+	if (!empty($conf->expedition->enabled)){
```
Comment (eldy) [inline htdocs/commande/list.php]: By default there is no warehouse on order so we should not enter this part of code by default.
May be you forget a test on the constant that enable the feature to manage warehouse on orders ?

Comment (bb2a) [reply htdocs/commande/list.php]: I add constant for enable code

**Comment by BB2A-Anthony** [comment]
is there another problem?

File: htdocs/commande/list.php [hunk 3]
```diff
@@ -588,7 +604,7 @@
 		$moreforfilter .= $formother->select_categories('customer', $search_categ_cus, 'search_categ_cus', 1);
 	 	$moreforfilter .= '</div>';
 	}
-	if (!empty($conf->expedition->enabled)){
+	if (!empty($conf->expedition->enabled) && $conf->global->WAREHOUSE_ASK_WAREHOUSE_DURING_ORDER = 1){
```
Comment (bb2a) [inline htdocs/commande/list.php]: Modif for conflict

**Comment by BB2A-Anthony** [comment]
@eldy Why is not merge?

File: htdocs/commande/list.php (hunk 3 above)
Comment (bb2a) [reply htdocs/commande/list.php]: @eldy its not a problem?

**Pull request merged**